src/app/app.component.ts

```diff
@@ -82,7 +88,21 @@ export class MyApp {
     // Reset the content nav to have just this page
     // we wouldn't want the back button to show in this scenario
     this.menuCtrl.close().then(() => {
-      this.nav.setRoot(page.component, { 'linkParam': page.param });
+      if(page.component === PushMessagesPage && !PushService.registered) {
+        this.alert.showAlert(constants.pushAlertMessage, constants.featureNotConfigured, 
+          constants.alertButtons, constants.showDocs, constants.pushDocsUrl);
+          return;
+      } else if (page.component === AuthPage) {
+        const configKeys: string[] = Object.keys(this.auth.getConfig());
+        // the following check is used to determine whether the Keycloak configuration
+        // and 'url' value in this configuration are present
+        if (configKeys.length <= 1 || configKeys.indexOf('url') === -1) {
```

Comment:
can we just check if `auth.getConfig()` returns `null` here?

Reply:
@wei-lee yes. Will do that.

Reply:
I don't think `Auth#getConfig` can ever return `null` at the moment, it appears to return an empty object instead. We could probably make a change to the SDK to return `null` if the config is not valid, or add a separate function `Auth#isConfigValid`

Reply:
@aidenkeating I am thinking about using the core module to look up the configuration directly?

IIRC, Auth#getConfig is for getting the authserviceConfig, not the configuration in mobile-services.json file. Am i right?

Reply:
@wei-lee Yep that looks to be correct. Only issue I can see with using the core directly is do we want to show the dialog only if the configuration object for a service doesn't exist at all?

For example, if the push configuration is missing the `variantId` field do we want to show the dialog, if so we should probably do that validation in the push SDK and provide a way for the end user to be notified. Or do we only want to show the dialog if the entire config object is missing from `mobile-services.json`?